### PR TITLE
Include Kptfiles and function configs in tree cmd

### PR DESCRIPTION
**Pull request by phanimarupaka**
This PR is to cleanup tree command. There are many flags which might not be used at all. We should be aggressive in cleaning up such flags and add them in future if there is a need. Or else we will be stuck with them forever. This PR also includes Kptfile and function configs as part of tree command output. Copies over the implementation of tree command to kpt from kustomize.

### CODE REVIEW

File: thirdparty/cmdconfig/commands/cmdtree/cmdtree.go [hunk 1]
```diff
@@ -83,107 +48,24 @@ func (r *TreeRunner) runE(c *cobra.Command, args []string) error {
...
 	return runner.HandleError(c, kio.Pipeline{
 		Inputs:  []kio.Reader{input},
 		Filters: fltrs,
-		Outputs: []kio.Writer{kio.TreeWriter{
+		Outputs: []kio.Writer{TreeWriter{
 			Root:            root,
 			Writer:          c.OutOrStdout(),
-			Fields:          fields,
-			Structure:       kio.TreeStructure(r.structure),
 			OpenAPIFileName: kptfilev1alpha2.KptFileName,
```
Comment (droot) [inline thirdparty/cmdconfig/commands/cmdtree/cmdtree.go:64]: I found `OpenAPIFileName` to be misleading here. I am guessing this was to enable parameterization of pkg file. We can probably remove this parameter and use the `Kptfile` directly ?

Comment (phanimarupaka) [reply thirdparty/cmdconfig/commands/cmdtree/cmdtree.go]: Good point. Done.

File: go.mod [hunk 2]
```diff
@@ -14,6 +14,7 @@ require (
 	github.com/spf13/cobra v1.1.1
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.6.1
+	github.com/xlab/treeprint v0.0.0-20181112141820-a009c3971eca
```
Comment (mortent) [inline go.mod:17]: Looks like there is a pretty recent release of this library: https://github.com/xlab/treeprint/releases/tag/v1.1.0. Can we use that?

File: thirdparty/cmdconfig/commands/cmdtree/tree.go [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+type TreeStructure string
+
+const (
+	// TreeStructurePackage configures TreeWriter to generate the tree structure off of the
+	// Resources packages.
+	TreeStructurePackage TreeStructure = "directory"
+
+	// TreeStructureOwners configures TreeWriter to generate the tree structure off of the
+	// Resource owners.
+	TreeStructureGraph TreeStructure = "owners"
```
Comment (mortent) [inline thirdparty/cmdconfig/commands/cmdtree/tree.go:29]: Do we ever use this? Configuration doesn't usually have owner refs before it is applied to the cluster.

File: go.mod (hunk 2 above)
Comment (phanimarupaka) [reply go.mod]: Thanks. Done.

File: thirdparty/cmdconfig/commands/cmdtree/tree.go (hunk 3 above)
Comment (phanimarupaka) [reply thirdparty/cmdconfig/commands/cmdtree/tree.go]: We don't currently. Cleaned up the code.

**Review by droot — APPROVED** [review]

**Pull request merged**